In [19]:
import numpy as np
import scipy as sc


1 )  Даны две  независимые выборки. Не соблюдается условие нормальности
x1  380,420, 290
y1 140,360,200,900
Сделайте вывод по результатам, полученным с помощью функции

In [24]:
x1 = np.array([380, 420, 290])
y1 = np.array([140, 360, 200, 900])
sc.stats.mannwhitneyu(x1, y1)

MannwhitneyuResult(statistic=8.0, pvalue=0.6285714285714286)

pvalue=0.6285714285714286 нет статистически значимых различий 

2 ) Исследовалось влияние препарата на уровень давления пациентов. Сначала измерялось давление до приема препарата, потом через 10 минут и через 30 минут. Есть ли статистически значимые различия?

1е измерение до приема препарата: 150, 160, 165, 145, 155
2е измерение через 10 минут: 140, 155, 150,  130, 135
3е измерение через 30 минут: 130, 130, 120, 130, 125

In [21]:
a = [150, 160, 165, 145, 155]
b = [140, 155, 150, 130, 135]
c = [130, 130, 120, 130, 125]

sc.stats.friedmanchisquare(a,b, c)

FriedmanchisquareResult(statistic=9.578947368421062, pvalue=0.00831683351100441)

pvalue=0.00831683351100441 есть статистически значимые различия

3 ) Сравните 1 и 2 е измерения, предполагая, что 3го измерения через 30 минут не было.

In [22]:
a = [150, 160, 165, 145, 155]
b = [140, 155, 150, 130, 135]
sc.stats.wilcoxon(a, b)

WilcoxonResult(statistic=0.0, pvalue=0.0625)

pvalue=0.0625 нет статистически значимых разлчий 


4) Даны 3 группы  учеников плавания.
В 1 группе время на дистанцию 50 м составляют:
56, 60, 62, 55, 71, 67, 59, 58, 64, 67

Вторая группа : 57, 58, 69, 48, 72, 70, 68, 71, 50, 53
Третья группа: 57, 67, 49, 48, 47, 55, 66, 51, 54

In [23]:
a1 = [56, 60, 62, 55, 71, 67, 59, 58, 64, 67]
b1 = [57, 58, 69, 48, 72, 70, 68, 71, 50, 53]
c1 = [57, 67, 49, 48, 47, 55, 66, 51, 54]

sc.stats.kruskal(a1, b1, c1)



KruskalResult(statistic=5.465564058257224, pvalue=0.06503809985904942)

pvalue=0.06503809985904942 нет статистически значимых различий 

5) Заявляется, что партия изготавливается со средним арифметическим 2,5 см. Проверить данную гипотезу, если известно, что размеры изделий подчинены нормальному закону распределения. Объем выборки 10, уровень статистической значимости 5%

2.51, 2.35, 2.74, 2.56, 2.40, 2.36, 2.65, 2.7, 2.67, 2.34

In [26]:
a = [2.51, 2.35, 2.74, 2.56, 2.40, 2.36, 2.65, 2.7, 2.67, 2.34]
sc.stats.ttest_1samp(a, 2.5)

TtestResult(statistic=0.5630613661802959, pvalue=0.5871439993940629, df=9)

In [38]:
import math
from scipy.stats import t
data = [2.51, 2.35, 2.74, 2.56, 2.40, 2.36, 2.65, 2.7, 2.67, 2.34]
mean = sum(data) / len(data)

In [42]:
s = math.sqrt(sum([(x - mean)**2 for x in data]) / (len(data) - 1))
x = (mean - 2.5) / (s / math.sqrt(len(data)))
p = (1 - t.cdf(abs(x), len(data) - 1)) * 2
print(p)

0.5871439993940628


pvalue=0.5871439993940629 нет статистически значимых различий 